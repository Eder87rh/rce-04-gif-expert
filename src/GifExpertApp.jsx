import React from "react";
import { useState } from "react";
import AddCategory from "./components/AddCategory";

const GifExpertApp = () => {
  const [categories, setCategories] = useState([
    "One punch",
    "Samurai X",
    "Dragon ball",
  ]);

  /* const handleAdd = (el) => {
    setCategories([...categories, el]);
  }; */

  return (
    <div>
      <h2>GifExpertApp</h2>
      <AddCategory />
      <hr />

      <ol>
        {categories.map((category) => {
          return <li key={category}>{category}</li>;
        })}
      </ol>
    </div>
  );
};

export default GifExpertApp;
