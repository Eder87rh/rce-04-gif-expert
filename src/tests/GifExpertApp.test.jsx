import { shallow } from "enzyme";
import GifExpertApp from "../GifExpertApp";

describe("GifExpertApp", () => {
  it("should render correctly", () => {
    const wrapper = shallow(<GifExpertApp />);
    expect(wrapper).toMatchSnapshot();
  });

  it('should show a list of categories', () => {
    const categories = ['One punch', 'Dragon ball'];
    const wrapper = shallow(<GifExpertApp defaultCategories={categories} />);

    expect(wrapper).toMatchSnapshot();
    expect(wrapper.find('GifGrid').length).toBe(categories.length);
  })
  
});
