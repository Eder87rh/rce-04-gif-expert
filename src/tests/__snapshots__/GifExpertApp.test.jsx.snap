// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GifExpertApp should render correctly 1`] = `
<div>
  <h2>
    GifExpertApp
  </h2>
  <AddCategory
    setCategories={[Function]}
  />
  <hr />
  <ol />
</div>
`;

exports[`GifExpertApp should show a list of categories 1`] = `
<div>
  <h2>
    GifExpertApp
  </h2>
  <AddCategory
    setCategories={[Function]}
  />
  <hr />
  <ol>
    <GifGrid
      category="One punch"
      key="One punch"
    />
    <GifGrid
      category="Dragon ball"
      key="Dragon ball"
    />
  </ol>
</div>
`;
